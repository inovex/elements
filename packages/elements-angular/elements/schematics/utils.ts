import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
