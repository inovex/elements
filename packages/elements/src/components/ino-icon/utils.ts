import { getAssetPath } from '@stencil/core';

let CACHED_MAP: Map<string, string>;

export function getIconMap(): Map<string, string> {
	if (!CACHED_MAP) {
		const win = window as any;
		win.inoIcons = win.inoIcons || {};
		CACHED_MAP = win.inoIcons.map = win.inoIcons.map || new Map();
	}
	return CACHED_MAP;
}

export function addIcons(icons: { [name: string]: string }) {
	const map = getIconMap();
	Object.keys(icons).forEach((name) => map.set(name, icons[name]));
}

export const getSrc = (src: string | undefined) => {
	if (isStr(src)) {
		src = src.trim();
		if (isSrc(src)) {
			return src;
		}
	}
	return null;
};

export function getName(name: string | undefined) {
	if (name && !isSrc(name)) {
		return name;
	}
	return undefined;
}

export function getUrl(src: string | undefined, icon: string | undefined) {
	let url = getSrc(src);

	if (url) {
		return url;
	}
	url = getName(icon);
	if (url) {
		return getNamedUrl(url);
	}

	url = getSrc(icon);
	if (url) {
		return url;
	}

	return null;
}

export function getNamedUrl(name: string) {
	const url = getIconMap().get(name);
	if (url) {
		return url;
	}
	return getAssetPath(`./ino-icon/${name}.svg`);
}

export const isSrc = (str: string) => str.length > 0 && /(\/|\.)/.test(str);

export const isStr = (val: any): val is string => typeof val === 'string';

export const toLower = (val: string) => val.toLowerCase();
