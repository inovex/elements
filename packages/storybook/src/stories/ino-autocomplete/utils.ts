import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
