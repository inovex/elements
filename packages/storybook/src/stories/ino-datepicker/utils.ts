import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
