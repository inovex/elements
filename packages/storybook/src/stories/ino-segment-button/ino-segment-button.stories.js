import componentReadme from '_local-elements/src/components/ino-segment-button/readme.md';
import withStencilReadme from '_local-storybookcore/with-stencil-readme';
import './ino-segment-button.scss';
import { boolean, text } from '@storybook/addon-knobs';

export default {
  title: 'Buttons/<ino-segment-button>',
  decorators: [withStencilReadme(componentReadme)],
};

export const DefaultUsage = () => /* html */ `
    <div class="story-segment-button">
        <style>
          ino-segment-button.customizable-segment-btn {
              --ino-segment-button-color: ${text('--ino-segment-button-color', '#c1c1c1', 'Custom Properties')};
              --ino-segment-button-checked-color: ${text('--ino-segment-button-checked-color', '#0d10f3', 'Custom Properties')};
              --ino-segment-button-hover-color: ${text('--ino-segment-button-hover-color', '#5d60f7', 'Custom Properties')};
              --ino-segment-button-disabled-color: ${text('--ino-segment-button-disabled-color', '#777777', 'Custom Properties')};
              --ino-segment-button-hover-shadow: ${text('--ino-segment-button-hover-shadow', '0 4px 8px 0 rgba(61, 64, 245, 0.3), 0 12px 28px 0 rgba(61, 214, 245, 0.14)', 'Custom Properties')};
              --ino-segment-button-checked-shadow: ${text('--ino-segment-button-checked-shadow', '0 8px 16px 0 rgba(61, 64, 245, 0.3),0 16px 32px 0 rgba(61, 214, 245, 0.14)', 'Custom Properties')};
          }
        </style>
        <h4>Customizable Segment Button</h4>
        <ino-segment-button
                value="1"
                class="customizable-segment-btn"
                checked="${boolean('checked', false, 'General')}"
                disabled="${boolean('disabled', false, 'General')}"
                ino-dense="${boolean('ino-dense', false, 'General')}"
            >
            Customizable Segment Button
        </ino-segment-button>
        <h4>Enabled, not checked</h4>
        <ino-segment-button>Content</ino-segment-button>

        <h4>Enabled, checked</h4>
        <ino-segment-button checked>Content</ino-segment-button>

        <h4>Disabled</h4>
        <ino-segment-button disabled>Content</ino-segment-button>

        <h4>Dense</h4>
        <ino-segment-button ino-dense>Content</ino-segment-button>
    </div>

  `;
