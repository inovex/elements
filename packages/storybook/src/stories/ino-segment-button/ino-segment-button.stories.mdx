import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-segment-button/readme.md";

<Meta title="Buttons/ino-segment-button/Documentation" />

<Description markdown={readme} />
