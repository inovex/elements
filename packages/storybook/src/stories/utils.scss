/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
