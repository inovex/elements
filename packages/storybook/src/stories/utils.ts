import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
