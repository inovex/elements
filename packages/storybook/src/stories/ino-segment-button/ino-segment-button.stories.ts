import { Components } from '@inovex.de/elements';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator } from '../utils';

import './ino-segment-button.scss';

export default {
  title: 'Buttons/ino-segment-button',
  component: 'ino-segment-button',
  decorators: [story => defaultDecorator(story, 'story-segment-button')],
  parameters: {
    actions: {
      handles: [
        'checkedChange',
      ],
    },
  },
};
export const Playground: Story<Components.InoSegmentButton> = (
  args
) => html`
  <ino-segment-button
    value="1"
    class="customizable-segment-btn"
    checked="${args.checked}"
    dense="${args.dense}"
    disabled="${args.disabled}"
    name="${args.name}"
    value="${args.value}"
  >
    Segment Button Label
  </ino-segment-button>
`;
Playground.args = {
  checked: false,
  dense: false,
  disabled: false,
  name: '',
  value: ''
}

export const EnableUnchecked = () => html`
  <ino-segment-button>Content</ino-segment-button>
`;
export const EnableChecked = () => html`
  <ino-segment-button checked>Content</ino-segment-button>
`;
export const Disabled = () => html`
  <ino-segment-button disabled>Content</ino-segment-button>
`;
export const Dense = () => html`
  <ino-segment-button dense>Content</ino-segment-button>
`;

  /*
  <style>
  ino-segment-button.customizable-segment-btn {
      --ino-segment-button-color: ${text(
        '--ino-segment-button-color',
        '#c1c1c1',
        'Custom Properties'
      )};
      --ino-segment-button-checked-color: ${text(
        '--ino-segment-button-checked-color',
        '#0d10f3',
        'Custom Properties'
      )};
      --ino-segment-button-hover-color: ${text(
        '--ino-segment-button-hover-color',
        '#5d60f7',
        'Custom Properties'
      )};
      --ino-segment-button-disabled-color: ${text(
        '--ino-segment-button-disabled-color',
        '#777777',
        'Custom Properties'
      )};
      --ino-segment-button-hover-shadow: ${text(
        '--ino-segment-button-hover-shadow',
        '0 4px 8px 0 rgba(61, 64, 245, 0.3), 0 12px 28px 0 rgba(61, 214, 245, 0.14)',
        'Custom Properties'
      )};
      --ino-segment-button-checked-shadow: ${text(
        '--ino-segment-button-checked-shadow',
        '0 8px 16px 0 rgba(61, 64, 245, 0.3),0 16px 32px 0 rgba(61, 214, 245, 0.14)',
        'Custom Properties'
      )};
  }
</style>
*/