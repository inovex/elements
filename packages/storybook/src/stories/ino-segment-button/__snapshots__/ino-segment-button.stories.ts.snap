// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons/ino-segment-button Dense smoke-test 1`] = `
<div id="root-inner">
  <div class="story-segment-button">
    <ino-segment-button value
                        class="customizable-segment-btn hydrated"
                        checked="false"
                        dense="true"
                        disabled="false"
                        name
    >
      Segment Button Label
    </ino-segment-button>
  </div>
</div>
`;

exports[`Buttons/ino-segment-button Disabled smoke-test 1`] = `
<div id="root-inner">
  <div class="story-segment-button">
    <ino-segment-button value
                        class="customizable-segment-btn ino-segment-button--disabled hydrated"
                        checked="false"
                        dense="false"
                        disabled="true"
                        name
    >
      Segment Button Label
    </ino-segment-button>
  </div>
</div>
`;

exports[`Buttons/ino-segment-button EnableChecked smoke-test 1`] = `
<div id="root-inner">
  <div class="story-segment-button">
    <ino-segment-button value
                        class="customizable-segment-btn ino-segment-button--checked hydrated"
                        checked
                        dense="false"
                        disabled="false"
                        name
    >
      Segment Button Label
    </ino-segment-button>
  </div>
</div>
`;

exports[`Buttons/ino-segment-button Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-segment-button">
    <ino-segment-button value
                        class="customizable-segment-btn hydrated"
                        checked="false"
                        dense="false"
                        disabled="false"
                        name
    >
      Segment Button Label
    </ino-segment-button>
  </div>
</div>
`;
